src/chain/bls_emu.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+//! Types emulating the BLS functionality until proper BLS lands
+use super::{delivery_group_size, NetworkEvent, ProofSet, SectionInfo};
+use crate::id::{FullId, PublicId};
+use parsec;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fmt,
+};
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct PublicKeySet {
+    sec_info: SectionInfo,
+    threshold: usize,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct PublicKey(PublicKeySet);
+
+pub type SignatureShare = ::safe_crypto::Signature;
+
+pub struct SecretKeyShare(FullId);
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct PublicKeyShare(PublicId);
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct Signature {
```

Comment:
This struct is supposed to emulate the BLS `Signature` type, hence it's singular - with BLS, this would be indeed just a single signature, here we just emulate the functionality using a set of signatures.
I could make it a tuple struct, though.

Reply:
Quite happy with the struct with a name as it is personally.